src/main/java/org/apache/commons/text/CaseUtils.java

```diff
@@ -96,11 +96,100 @@ public static String toCamelCase(String str, final boolean capitalizeFirstLetter
         return new String(newCodePoints, 0, outOffset);
     }
 
+    private static String toDelimiterCase(String str, final char newDelimiter, final char... delimiters) {
+        if (StringUtils.isEmpty(str)) {
+            return str;
+        }
+        str = str.toLowerCase();
+        final int strLen = str.length();
+        final int[] newCodePoints = new int[strLen];
```

Comment:
The number of code points might not the same as the number of chars.

Reply:
I'm no expert in Unicode, please see my understanding is correct or not:

Change to `final int[] newCodePoints = new int[str.codePointCount(0, strLen)];`.  But since code points <= number of chars, there will be no test case to validate this change. Also will change `toCamelCase()` as well.